src/support/file.cpp

```diff
@@ -24,24 +24,33 @@
 template <typename T>
 T wasm::read_file(const std::string &filename, Flags::BinaryOption binary, Flags::DebugOption debug) {
   if (debug == Flags::Debug) std::cerr << "Loading '" << filename << "'..." << std::endl;
-  std::ifstream infile;
-  std::ios_base::openmode flags = std::ifstream::in;
-  if (binary == Flags::Binary) flags |= std::ifstream::binary;
-  infile.open(filename, flags);
-  if (!infile.is_open()) {
+  FILE *handle = fopen(filename.c_str(), binary == Flags::Binary ? "rb" : "r");
```

Comment:
The issue was that I could not find a suitable way to get the number of chars that were actually read using a C++ API. However with https://github.com/WebAssembly/binaryen/pull/986#issuecomment-314779011 re-did this PR to use `istream::gcount()`, which gives that information.